### PR TITLE
Add a new test to verify python311-pipx

### PR DESCRIPTION
**Pull request by DeepthiYV**
- Related ticket: https://progress.opensuse.org/issues/156472
- Needles: No
- Verification run: 
  SLE15-SP6:  [aarch64](https://openqa.suse.de/tests/14948447) | [s390x](https://openqa.suse.de/tests/14948448) | [x86_64](https://openqa.suse.de/tests/14948449)
SLE15-SP5: [aarch64](https://openqa.suse.de/tests/14948441) | [s390x](https://openqa.suse.de/tests/14948442) | [x86_64](https://openqa.suse.de/tests/14948443)
SLE15-SP4: [aarch64](https://openqa.suse.de/tests/14948444) | [s390x](https://openqa.suse.de/tests/14948445) | [x86_64](https://openqa.suse.de/tests/14948446)
LEAP15.6: [aarch64](https://openqa.opensuse.org/tests/4367120#step/python3_pipx/1) | [ppc64le](https://openqa.opensuse.org/tests/4367119#step/python3_pipx/1)| [x86_64](https://openqa.opensuse.org/tests/4367118#step/python3_pipx/1)
Tumbleweed: https://openqa.opensuse.org/tests/4347165

### CODE REVIEW

File: tests/console/python3_pipx.pm [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    run_tests($system_python_version);
+    # Test all available new python3 versions in SLEs if any
+    if (is_sle() || is_leap('>15.5')) { run_tests($_) foreach (get_available_python_versions()); }
+}
+
+sub run_tests ($python3_spec_release) {
+    record_info("$python3_spec_release");
+    # Install pipx and wheel
+    if ((zypper_call("se -x $python3_spec_release-pipx", exitcode => [0, 104]) == 104) or (zypper_call("se -x $python3_spec_release-wheel", exitcode => [0, 104]) == 104)) {
+        record_info("Skip! either $python3_spec_release-pipx or $python3_spec_release-wheel doesn't exist");
```
Comment (rfan1) [inline tests/console/python3_pipx.pm:43]:
```suggestion
        record_info("Skip!", "either $python3_spec_release-pipx or $python3_spec_release-wheel doesn't exist");
```

File: tests/console/python3_pipx.pm [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    assert_script_run("export PATH=\$PATH:~/.local/bin");
+    validate_script_output("hello-world", sub { m/Hello world from package!/ });
+    validate_script_output("pipx uninstall package", sub { m/uninstalled package!/ });
+    validate_script_output("pipx list", sub { m/nothing has been installed with pipx/ });
+}
+
+sub cleanup {
+    if (is_sle() || is_leap('>15.5')) {
+        foreach my $python_version (get_available_python_versions()) {
+            zypper_call("rm $python_version-base", exitcode => [0, 104]);
```
Comment (dzedro) [inline tests/console/python3_pipx.pm:61]: Why `$python_version-base` ?

Comment (DeepthiYV) [reply tests/console/python3_pipx.pm:61]: Installing pipx also installs its dependency, python-base, along with some packages. Removing python-base will remove pipx, the wheel of a particular Python version, and its dependency packages.

Comment (dzedro) [reply tests/console/python3_pipx.pm:61]: ok, thanks

File: tests/console/python3_pipx.pm [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    assert_script_run("pipx install dist/package-0.1-py3-none-any.whl --python=/usr/bin/$python_binary");
+    script_run("pipx list");
+    assert_script_run("export PATH=\$PATH:~/.local/bin");
+    validate_script_output("hello-world", sub { m/Hello world from package!/ });
+    validate_script_output("pipx uninstall package", sub { m/uninstalled package!/ });
+    validate_script_output("pipx list", sub { m/nothing has been installed with pipx/ });
+}
+
+sub cleanup {
+    if (is_sle() || is_leap('>15.5')) {
```
Comment (dzedro) [inline tests/console/python3_pipx.pm:59]: Why is this executed only here ?

Comment (DeepthiYV) [reply tests/console/python3_pipx.pm:59]: The `cleanup` module is called in both the `post_run_hook` and `post_fail_hook` functions.

Comment (DeepthiYV) [reply tests/console/python3_pipx.pm:59]: Testing is done for the default-installed python/python_pipx in Tumbleweed, so no cleanup is required.

Comment (dzedro) [reply tests/console/python3_pipx.pm:59]: Hmm, ok.

**Review by dzedro — APPROVED** [review]

**Pull request merged**